{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[<IonQSimulatorBackend('ionq_simulator')>, <IonQQPUBackend('ionq_qpu')>]\n"
     ]
    }
   ],
   "source": [
    "# create bankend for Qiskit's AER (QSAM simulator)\n",
    "from qiskit import Aer\n",
    "aer_sim = Aer.get_backend('qasm_simulator')\n",
    "import os\n",
    "import threading\n",
    "ionq_token=\"LAYq6EwdvXqh4sIma2iKZNQ2l7KzbxX6\"\n",
    "from qiskit_ionq import IonQProvider\n",
    "from qiskit import QuantumCircuit, execute\n",
    "from threading import Thread, Lock\n",
    "provider = IonQProvider(ionq_token)\n",
    "lock = Lock()\n",
    "\n",
    "# show supported backends\n",
    "print(provider.backends()) \n",
    "\n",
    "# create backends for simulator and hardware\n",
    "backend_sim = provider.get_backend(\"ionq_simulator\")\n",
    "backend_qpu = provider.get_backend(\"ionq_qpu\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAKAAAAB7CAYAAADzNovBAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjMsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+AADFEAAAI8klEQVR4nO3df0yTiR3H8Y8FtR4gx8liDwp4CjLsRBTuRKcgObdzccsQUDSKmUBERFSUyKLy3wGekAgJycwRmKhxUUBHNFmMONqMQ0/ElYmaMDciP0YiOnb+4IcZ7f5gcldAKKbl+wCfV9Kk1PZ5vso7z/O0tk9nmM1mM4iEqKQHoOmNAZIoBkiiGCCJYoAkigGSKAZIohggiWKAJIoBkigGSKIYIIligCSKAZIoBkiiGCCJYoAkigGSKAZIohggiWKAJIoBkigGSKIYIIligCSKAZIoR+kBpFy+C7R3yazb0w2ICnm/xx48eBBGo9G2A1khKCgI+fn5Nl/utA2wvQv4x1PpKcbPaDTCYDBIj2Ez3AWTKAZIohggiWKAJIoBkigGSKIYIA3j6uo6Yeuatq8DTgceHh6Ijo5GSEgIAgICoFar0d3djQcPHqCurg4VFRXo7Oy0eIxGo4Fer8e1a9eQnp5u9xkZ4BTk6+uLnJwcREZGwtFx+K945cqViI+PR0FBAS5duoSjR4+itbV1MD5/f3/09PTAyckJr1+/tuusitsFm0wm5OXlwc/PD2q1GsuWLYPBYIC/vz92794tNlf5l+tw549fWn27lH379qGhoQExMTEwm80oLy9HcnIyVq9ejcDAQKxduxapqam4evUqHB0dsWPHDjQ2NmL//v2D8RmNRqxfv97u8QEK3ALGx8fjypUryMzMRHBwMGpra7Ft2zZ0dnbi0KFD0uMpWm5u7uBu89y5c8jIyEBHR8ew+9XU1KCwsBA+Pj44deoUNm3ahIKCAgAYjO/58+cTMrOiArxw4QJKS0uh1+sRHh4OAIiIiMC9e/dw+fJlBAcHC0+oXAcOHEB6ejrevHmDnTt34uLFi2M+5smTJ9i7dy9WrVoFjUYDACgqKpqw+ACF7YJzcnKwYcOGwfje8vX1xcyZM7F06VKhyZTN398fJ06cAABs377dqviA759waDQatLa2AgCys7Ph6elpt1mHUkyAbW1taGxsxObNm4f9WUtLC3Q6HWbPnj3mcmbMmGHVxWDQj3vGO5VZ+N3uDy0u/2qqGfdyDAa91XMOn3v4O2Fyc3OhVqtRUlKC8vJyq2b44RMOo9GI5cuXo7KyEq6ursjKyhphZsO45rSWogIEMLgreKunpwcGg0ERu9/Pfn0MyV//x+LisXiN6EwLFizAxo0b0dfXh4yMDKseMzS+t8d8aWlpMJlMiI2Nxbx58+w8+QDFBOju7g4AaGpqsrj95MmT6OjowIoVK6xajtlstuoSHr7O1n8Fq4WHr7N6zuFzWx6ebNmyBSqVCuXl5Xj27NmY635XfADQ3NyM69evQ61WIzIycsjM4eOa01qKCXDhwoUIDAxEdnY2zp49i5s3byI5ORklJSUAoIgtoBKFhAy8tfrGjRtj3ne0+N6qqqqyWK69KSZAlUqFsrIy6HQ6JCcnY9euXXB3d0dKSgocHR0RGBgoPaIiLVmyBADQ0NAw6v2siQ/A4Nv9dTqd7YcdgaJehlm8eDGqq6stbouLi0NAQADmzJkjNNWAmOP6cd0+UU6fPg2NRoOWlpZR71dUVDRmfMDAIVBWVhaam5vtMe4wigpwJHfv3kVoaKj0GIpVWFho1f327NmDgoICJCUljfo6X1tbG44fP26r8cakmF3wSF69eoWmpiarn4DQu7W3tyMmJmZCX2S2hqK3gM7Ozujv75ceg+xI0VtAmvoYIIligCSKAZIoBkiiGCCJYoAkStGvA9qTp9vkXHdQUNC4H/PPloG35S/0/tjiur3Xa40Z5vG8d4Ympd9+9TUA4ETGbovrSsBdMIligCSKAZIoBkiiGCCJYoAkigGSKAZIohggiWKAJIoBkigGSKIYIIligCRqWgWYmpoKrVY74om7aWR6vR46nQ6+vr5ITEy0+ee0p1WAsbGxqK+vlx5j0jCZTEhMTERZWRkeP36MFy9e4Pz58zZdx7QKcM2aNZg/f770GJNGXV0dPDw8Bs/AlZCQgIqKCpuuY1oFSOPT1tYGLy+vwZ+9vb0HzyVtKzwYmoKedX2HC5VVGPphi4LfVwy77uCgws5NP8dcF6dhyzGbzRbne7bHpze4BZyC3N1c4bdAi46nz9Hx9PuzYQ293vH0OX6y+JMR4wMALy8vi/MOtra2QqvV2nRWBjhF/WxNCDQ/+mjU+yzQahD22bvPPBsSEoL29nY8fPgQAFBcXIyoqCibzjmtAkxKSoJWq0V/fz+0Wi1SUlKkR7IbR0cHxP4yAg4OI/+KZ82aiS0b10GlencCDg4OKCoqQkxMDBYtWgRnZ2fExcXZdE5+LPP/hh7vTBWGbxvwJ/23w26P/kUYPg38scBElqbVFnA0f771V1yorEJ/v0l6FJta++lSfOJl+SH0AF8fhCz1F5rIEgME0NPbh7/c+Rv+29//zl3WZKVSqbB54zrMnjUTAOD0gRrRG8IUs7VX7L/2/fv3ER0dDXd3d6jVavj5+eHYsWN2Wdc39Y3o7XuDz386Nb+L5CNXF/zq89UAgOgNYXB2kv3GgR9S5DFgfX09wsLC4OPjgyNHjsDb2xvNzc2ora1FcXHxqI99e+oJkmXtqT8U+UL04cOH4eLigtu3b2Pu3LmDtyckJAhORfaguC1gd3c3XFxckJqaivz8fLuuq6e3D1+d/gMWen+MnVFf2HVdNDLFbQG7urpgMpne+ztr32cX/PDvT7jrtjFrd8GKexLi5uYGlUqF9vZ26VFoAihuFwwAERERePToEZqamiyOAW2p6pt6VNXUI/U3UfCc726XddDYFLcFBIC8vDy8fPkSoaGhOHPmDKqrq1FaWorExESbLL+ntw81dfexxM+H8QlT3DEgMPDdwLdu3UJmZibS0tLQ29sLLy8vbN261SbL//d3L/HBnNlT9nW/yUSRu+CJYDKZRv2PeJoY0zZAUgZuAkgUAyRRDJBEMUASxQBJFAMkUQyQRDFAEsUASRQDJFEMkEQxQBLFAEkUAyRRDJBEMUASxQBJFAMkUQyQRDFAEsUASRQDJFEMkEQxQBLFAEkUAyRRDJBE/Q+Eux4n6re6MAAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 193.726x144.48 with 1 Axes>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from qiskit import QuantumCircuit\n",
    "\n",
    "qc = QuantumCircuit(1,1)\n",
    "qc.h(0)\n",
    "qc.measure(0,0)\n",
    "\n",
    "qc.draw(output='mpl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def gen_1000_bits(): \n",
    "    qpu_job = execute(qc, backend = backend_qpu, shots = 1000, memory = True)\n",
    "    results = qpu_job.result()\n",
    "    memory = results.get_memory(qc)\n",
    "    \n",
    "    lock.acquire()\n",
    "    bits = open(\"bits3.txt\", \"a\")\n",
    "    for bit in memory:\n",
    "        bits.write(bit)\n",
    "    lock.free()\n",
    "    \n",
    "    return memory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-17:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 237, in get_memory\n",
      "    memory = self.data(experiment)[\"memory\"]\n",
      "KeyError: 'memory'\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-5-0b4e67f327f1>\", line 4, in gen_1000_bits\n",
      "    memory = results.get_memory(qc)\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 254, in get_memory\n",
      "    ) from ex\n",
      "qiskit.exceptions.QiskitError: 'No memory for experiment \"<qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x7fd6fb53d990>\". Please verify that you either ran a measurement level 2 job with the memory flag set, eg., \"memory=True\", or a measurement level 0/1 job.'\n",
      "\n",
      "Exception in thread Thread-14:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 237, in get_memory\n",
      "    memory = self.data(experiment)[\"memory\"]\n",
      "KeyError: 'memory'\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-5-0b4e67f327f1>\", line 4, in gen_1000_bits\n",
      "    memory = results.get_memory(qc)\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 254, in get_memory\n",
      "    ) from ex\n",
      "qiskit.exceptions.QiskitError: 'No memory for experiment \"<qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x7fd6fb53d990>\". Please verify that you either ran a measurement level 2 job with the memory flag set, eg., \"memory=True\", or a measurement level 0/1 job.'\n",
      "\n",
      "Exception in thread Thread-16:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 237, in get_memory\n",
      "    memory = self.data(experiment)[\"memory\"]\n",
      "KeyError: 'memory'\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-5-0b4e67f327f1>\", line 4, in gen_1000_bits\n",
      "    memory = results.get_memory(qc)\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 254, in get_memory\n",
      "    ) from ex\n",
      "qiskit.exceptions.QiskitError: 'No memory for experiment \"<qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x7fd6fb53d990>\". Please verify that you either ran a measurement level 2 job with the memory flag set, eg., \"memory=True\", or a measurement level 0/1 job.'\n",
      "\n",
      "Exception in thread Thread-22:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 237, in get_memory\n",
      "    memory = self.data(experiment)[\"memory\"]\n",
      "KeyError: 'memory'\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-5-0b4e67f327f1>\", line 4, in gen_1000_bits\n",
      "    memory = results.get_memory(qc)\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 254, in get_memory\n",
      "    ) from ex\n",
      "qiskit.exceptions.QiskitError: 'No memory for experiment \"<qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x7fd6fb53d990>\". Please verify that you either ran a measurement level 2 job with the memory flag set, eg., \"memory=True\", or a measurement level 0/1 job.'\n",
      "\n",
      "Exception in thread Thread-15:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 237, in get_memory\n",
      "    memory = self.data(experiment)[\"memory\"]\n",
      "KeyError: 'memory'\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-5-0b4e67f327f1>\", line 4, in gen_1000_bits\n",
      "    memory = results.get_memory(qc)\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 254, in get_memory\n",
      "    ) from ex\n",
      "qiskit.exceptions.QiskitError: 'No memory for experiment \"<qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x7fd6fb53d990>\". Please verify that you either ran a measurement level 2 job with the memory flag set, eg., \"memory=True\", or a measurement level 0/1 job.'\n",
      "\n",
      "Exception in thread Thread-23:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 237, in get_memory\n",
      "    memory = self.data(experiment)[\"memory\"]\n",
      "KeyError: 'memory'\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-5-0b4e67f327f1>\", line 4, in gen_1000_bits\n",
      "    memory = results.get_memory(qc)\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 254, in get_memory\n",
      "    ) from ex\n",
      "qiskit.exceptions.QiskitError: 'No memory for experiment \"<qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x7fd6fb53d990>\". Please verify that you either ran a measurement level 2 job with the memory flag set, eg., \"memory=True\", or a measurement level 0/1 job.'\n",
      "\n",
      "Exception in thread Thread-21:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 237, in get_memory\n",
      "    memory = self.data(experiment)[\"memory\"]\n",
      "KeyError: 'memory'\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-5-0b4e67f327f1>\", line 4, in gen_1000_bits\n",
      "    memory = results.get_memory(qc)\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 254, in get_memory\n",
      "    ) from ex\n",
      "qiskit.exceptions.QiskitError: 'No memory for experiment \"<qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x7fd6fb53d990>\". Please verify that you either ran a measurement level 2 job with the memory flag set, eg., \"memory=True\", or a measurement level 0/1 job.'\n",
      "\n",
      "Exception in thread Thread-20:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 237, in get_memory\n",
      "    memory = self.data(experiment)[\"memory\"]\n",
      "KeyError: 'memory'\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-5-0b4e67f327f1>\", line 4, in gen_1000_bits\n",
      "    memory = results.get_memory(qc)\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 254, in get_memory\n",
      "    ) from ex\n",
      "qiskit.exceptions.QiskitError: 'No memory for experiment \"<qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x7fd6fb53d990>\". Please verify that you either ran a measurement level 2 job with the memory flag set, eg., \"memory=True\", or a measurement level 0/1 job.'\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-19:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 237, in get_memory\n",
      "    memory = self.data(experiment)[\"memory\"]\n",
      "KeyError: 'memory'\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-5-0b4e67f327f1>\", line 4, in gen_1000_bits\n",
      "    memory = results.get_memory(qc)\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 254, in get_memory\n",
      "    ) from ex\n",
      "qiskit.exceptions.QiskitError: 'No memory for experiment \"<qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x7fd6fb53d990>\". Please verify that you either ran a measurement level 2 job with the memory flag set, eg., \"memory=True\", or a measurement level 0/1 job.'\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Time to generate 10000 bits using 10000 shots: 436.8825490474701 seconds\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-18:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 237, in get_memory\n",
      "    memory = self.data(experiment)[\"memory\"]\n",
      "KeyError: 'memory'\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-5-0b4e67f327f1>\", line 4, in gen_1000_bits\n",
      "    memory = results.get_memory(qc)\n",
      "  File \"/Users/mcbookair/opt/anaconda3/lib/python3.7/site-packages/qiskit/result/result.py\", line 254, in get_memory\n",
      "    ) from ex\n",
      "qiskit.exceptions.QiskitError: 'No memory for experiment \"<qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x7fd6fb53d990>\". Please verify that you either ran a measurement level 2 job with the memory flag set, eg., \"memory=True\", or a measurement level 0/1 job.'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import time\n",
    "start = time.time()\n",
    "\n",
    "t1 = Thread(target = gen_1000_bits)\n",
    "t2 = Thread(target = gen_1000_bits)\n",
    "t3 = Thread(target = gen_1000_bits)\n",
    "t4 = Thread(target = gen_1000_bits)\n",
    "t5 = Thread(target = gen_1000_bits)\n",
    "t6 = Thread(target = gen_1000_bits)\n",
    "t7 = Thread(target = gen_1000_bits)\n",
    "t8 = Thread(target = gen_1000_bits)\n",
    "t9 = Thread(target = gen_1000_bits)\n",
    "t10 = Thread(target = gen_1000_bits)\n",
    "\n",
    "t1.start()\n",
    "t2.start()\n",
    "t3.start()\n",
    "t4.start()\n",
    "t5.start()\n",
    "t6.start()\n",
    "t7.start()\n",
    "t8.start()\n",
    "t9.start()\n",
    "t10.start()\n",
    "\n",
    "\n",
    "t1.join()\n",
    "t2.join()\n",
    "t3.join()\n",
    "t4.join()\n",
    "t5.join()\n",
    "t6.join()\n",
    "t7.join()\n",
    "t8.join()\n",
    "t9.join()\n",
    "t10.join()\n",
    "\n",
    "time = time.time() - start\n",
    "\n",
    "print(f\"Time to generate 10000 bits using 10000 shots: {time} seconds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
